geonode/base/auth.py

```diff
@@ -105,8 +105,9 @@ def get_auth_user(access_token, client=settings.OAUTH2_DEFAULT_BACKEND_CLIENT_NA
     try:
         Application = get_application_model()
         app = Application.objects.get(name=client)
-        user = AccessToken.objects.filter(token=access_token, application=app).order_by('-expires').first().user
-        return user
+        access_token = AccessToken.objects.filter(token=access_token, application=app).order_by('-expires').first()
+        if access_token.is_valid():
```

Comment:
in case the `access_token` does not exist, the `first()` will return a None.
To make it safer is better to validate that the token exists, like:

```python
if access_token and access_token.is_valid():
    # do stuff
```

Reply:
done